Improve the Configure plugins page and document Fleet management (#2627)
neo4j/docs-operations: modules/ROOT/pages/configuration/plugins.adoc

= Configure plugins
:description: This page describes how to load plugins to a Neo4j deployment.

== Overview

Plugins are Java Archive (_.jar_) files that extend the functionality of Neo4j by adding new features and capabilities, such as graph algorithms, data integration, and visualization tools.

> Neo4j Community Edition (CE) comes with a range of pre-installed products, such as Gen AI, Graph Data Science, and Fleet management in the _products_ directory and the APOC Core jar file in the _labs_ directory.

Reviewer: It seems that CE doesn't come with a bundled GDS plugin:

> On a self-managed Neo4j Server you need to install and configure GDS manually.
> 
> * If you have the Neo4j Enterprise Edition, you will find the GDS plugin bundled within the Neo4j Server distribution. The GDS plugin is located in the products directory. This is the [Bundled installation](https://neo4j.com/docs/graph-data-science/current/installation/neo4j-server/#installation-neo4j-server-bundled) option.
> 
> * If you are using the Neo4j Community Edition, **you must download GDS manually from the** [Neo4j Deployment Center](https://neo4j.com/deployment-center/#gds-tab). This is the [Custom installation](https://neo4j.com/docs/graph-data-science/current/installation/neo4j-server/#installation-neo4j-server-custom) option.

See https://neo4j.com/docs/graph-data-science/current/installation/neo4j-server/#installation-neo4j-server-custom.

> Neo4j Community Edition (CE) comes with a range of pre-installed products, such as Gen AI, Graph Data Science, and Fleet management in the _products_ directory and the APOC Core jar file in the _labs_ directory.

Reviewer: hmm, interesting. You're right. CE contains the GDS plugin. But at the same time, if you visit [the Deployment centre](https://neo4j.com/deployment-center/#ga-tab), you'll read: "Neo4j Community Edition package and the open source Community Edition of the Graph Data Science Library are available as separate downloads."

<img width="1105" height="559" alt="Screenshot 2025-09-30 at 17 19 25" src="https://github.com/user-attachments/assets/d9ab4ec7-6600-4582-93dd-faf0f8adfccb" />

I'm totally confused. I asked Nicola about all GDS plugins

The Fleet management jar file is bundled with the Neo4j server starting with Neo4j 5.26.10 and 2025.07.0 onwards.
If you are using an earlier version of Neo4j, you can get the matching version of the Fleet management plugin by following the download link that matches your server version in the *Monitor Deployment* wizard in the Aura console.
See link:{neo4j-docs-base-uri}/aura/fleet-management/setup/[Aura documentation -> Fleet Management -> Add a deployment] for more information.

[NOTE]
====
Using the Fleet management plugin in Neo4j CE deployments is subject to some limitations due to the limited set of features, such as no metrics being available.
Also, it is offered on a best-effort basis.
Neo4j does not offer any technical support for it.
====

Some of these plugins, such as Bloom and GDS Enterprise require a license activation key.
Reach out to your Neo4j account representative or request a representative to link:https://neo4j.com/contact-us/#sales-inquiry[contact you].

> If you want to use your own plugins or any of the other supported plugins, such as Apoc Extended and Neo4j Semantics, ensure that you download and add them to the _plugins_ directory.

Reviewer: ```suggestion
If you want to use your own plugins or any of the other supported plugins, such as APOC Extended and Neosemantics, ensure that you download and add them to the _plugins_ directory.
```

See their respective documentation for more information.

[[supported-plugins]]
== Supported plugins and documentation

The following plugins are supported:

.Supported Neo4j plugins and documentation
[options="header",cols="d,m,b,a"]
|===
|Name |Key  | License activation key required | Documentation

| APOC Core
| `apoc`
|
| link:https://neo4j.com/docs/apoc/current/[APOC Core documentation]

| APOC Extended
| `apoc-extended`
|

> | link:https://neo4j.com/labs/apoc/5/[APOC Extended user guide (Labs project)]

Reviewer: There is also APOC Extended for 2025 -- https://neo4j.com/labs/apoc/2025/overview/. Should we link to them?

Author: Ah, right!

| Bloom
| `bloom`
| {check-mark} footnote:[You can also get basic access, without a license key, to link:{neo4j-docs-base-uri}/bloom-user-guide/current/bloom-installation/bloom-deployment-modes/[Bloom via Graph Apps] in Neo4j Desktop or the link:http://console-preview.neo4j.io/self-managed[Neo4j Aura console].]
| link:{neo4j-docs-base-uri}/bloom-user-guide[Neo4j Bloom]

| Fleet management
| `fleet-management`
|
| link:{neo4j-docs-base-uri}/aura/fleet-management/setup/[Fleet management]

| GenAI
| `genai`
|
| link:{neo4j-docs-base-uri}/cypher-manual/current/genai-integrations/[Cypher Manual -> GenAI integrations].

| Graph Data Science
| `graph-data-science`
| {check-mark} footnote:[Graph Data Science is available in both Neo4j CE and EE editions. The Enterprise Edition includes additional features and requires a license key.]
| link:{neo4j-docs-base-uri}/graph-data-science/current/installation/#_installation_methods[Graph Data Science]

> | Neo Semantics

Reviewer: ```suggestion
| Neosemantics
```

| `n10s`
|

> | link:https://neo4j.com/labs/neosemantics/[Neo Semantics]

Reviewer: ```suggestion
| link:https://neo4j.com/labs/neosemantics/[Neosemantics]
```

== Install and configure plugins

To install and configure plugins in a Neo4j deployment, follow these steps:

. Move or copy the plugins (_.jar files_) from _<NEO4J_HOME>/products_ and _<NEO4J_HOME>/labs_ to the _<NEO4J_HOME>/plugins_ directory.
See xref:configuration/file-locations.adoc[Default file locations] for more information.
+
[NOTE]
====

> Some plugins are not bundled with Neo4j and need to be downloaded separately, such as APOC Extended and Neo Semantics.

Reviewer: ```suggestion
Some plugins are not bundled with Neo4j and need to be downloaded separately, such as APOC Extended and Neosemantics.
```

See <<supported-plugins, Supported plugins and documentation>> for more information and links to the respective documentation.
====

. Configure the plugins that you want to enable by modifying the following setting to the _neo4j.conf_ file:

* Add the plugin that you want to enable to the xref:configuration/configuration-settings.adoc#config_dbms.security.procedures.unrestricted[`dbms.security.procedures.unrestricted`] setting.
* If the plugin requires a license key, add the path to the license key file as well.
* If xref:configuration/configuration-settings.adoc#config_dbms.security.procedures.allowlist[`dbms.security.procedures.allowlist`] is set in your configuration, also add the plugin to this setting, otherwise no change is needed.
By default, all procedures are loaded.
For more information, see xref:security/securing-extensions.adoc[Securing extensions].
+
[source, properties]
----

> # to enable APOC:

Reviewer: ```suggestion
# to enable APOC Core:
```

# * dbms.security.procedures.unrestricted=apoc.*
# * dbms.security.procedures.allowlist=apoc.*

# to enable APOC Extended:
# * dbms.security.procedures.unrestricted=apoc.*,apoc.extended.*
# * dbms.security.procedures.allowlist=apoc.*,apoc.extended.*

#  to enable Bloom:
# * dbms.security.procedures.unrestricted=bloom.*
# * dbms.security.procedures.allowlist=bloom.*
# * dbms.bloom.license_file=/path/to/my/license/keyfile

# to enable Fleet management:
# * dbms.security.procedures.unrestricted=fleetManagement.*
# * dbms.security.procedures.allowlist=fleetManagement.*

# to enable GDS:
# * dbms.security.procedures.unrestricted=gds.*
# * dbms.security.procedures.allowlist=gds.*
# * gds.enterprise.license_file=/path/to/my/license/keyfile

#  to enable both GDS and Bloom:
# * dbms.security.procedures.unrestricted=gds.*,bloom.*
# * dbms.security.procedures.allowlist=gds.*,bloom.*
# * gds.enterprise.license_file=/path/to/my/license/keyfile
# * dbms.bloom.license_file=/path/to/my/license/keyfile

# to enable GenAI:
# * dbms.security.procedures.unrestricted=genai.*
# * dbms.security.procedures.allowlist=genai.*

> # to enable Neo Semantics:

Reviewer: ```suggestion
# to enable Neosemantics:
```

# * dbms.security.procedures.unrestricted=n10s.*
# * dbms.security.procedures.allowlist=n10s.*
----
+
For more information on configuring the plugins, see the respective documentation:
+

> * link:https://neo4j.com/docs/apoc/current/installation/[APOC documentation]

Reviewer: ```suggestion
* link:https://neo4j.com/docs/apoc/current/installation/[APOC Core documentation]
```

> * link:https://neo4j.com/labs/apoc/5/installation/[APOC extended documentation]

Reviewer: or https://neo4j.com/labs/apoc/2025/overview/?

* link:https://neo4j.com/docs/bloom-user-guide/current/bloom-installation/[Bloom documentation]
* link:https://neo4j.com/docs/aura/fleet-management/setup/[Fleet management documentation]
* link:https://neo4j.com/docs/graph-data-science/current/installation/neo4j-server/[GDS documentation]
* link:https://neo4j.com/docs/cypher-manual/current/genai-integrations/[GenAI documentation]

> * link:https://neo4j.com/labs/neosemantics/[Neo Semantics documentation]

Reviewer: ```suggestion
* link:https://neo4j.com/labs/neosemantics/[Neosemantics documentation]
```

# * dbms.security.procedures.unrestricted=fleetManagement.*
# * dbms.security.procedures.allowlist=fleetManagement.*

# to enable GDS:
# * dbms.security.procedures.unrestricted=gds.*
# * dbms.security.procedures.allowlist=gds.*
# * gds.enterprise.license_file=/path/to/my/license/keyfile

#  to enable both GDS and Bloom:
# * dbms.security.procedures.unrestricted=gds.*,bloom.*
# * dbms.security.procedures.allowlist=gds.*,bloom.*
# * gds.enterprise.license_file=/path/to/my/license/keyfile
# * dbms.bloom.license_file=/path/to/my/license/keyfile

# to enable GenAI:
# * dbms.security.procedures.unrestricted=genai.*
# * dbms.security.procedures.allowlist=genai.*

# to enable GraphQL:
# * dbms.security.procedures.unrestricted=graphql.*
# * dbms.security.procedures.allowlist=graphql.*

# to enable Neo Semantics:
# * dbms.security.procedures.unrestricted=n10s.*
# * dbms.security.procedures.allowlist=n10s.*
----
+
For more information on configuring the plugins, see the respective documentation:
+
* link:https://neo4j.com/docs/bloom-user-guide/current/bloom-installation/[Bloom documentation]
* link:https://neo4j.com/docs/graph-data-science/current/installation/[GDS documentation]

> * link:https://neo4j.com/docs/aura/fleet-management/setup/[Fleet management documentation]

Reviewer 2: Can't verify if this is a valid link since the docs preview didn't work for this PR.
